functional-programming-spreadsheet/index350.js

```diff
@@ -0,0 +1,46 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const spreadsheetFunctions = {
+  "": x => x
+};
+
+const applyFn = str => {
+  const noHigh = highPrecedence(str);
+  const infix = /([0-9.]+)([+-])([0-9.]+)/;
+  const str2 = infixEval(noHigh, infix);
+  const regex = /([a-z]*)\(([0-9., ]*)\)(?!.*\()/i;
+  const toNumberList = args => args.split(",").map(parseFloat);
+  const applyFunction = (fn, args) =>
+    spreadsheetFunctions[fn.toLowerCase()](toNumberList(args));
+  return str2.replace(
+    regex,
+    (match, fn, args) =>
+      spreadsheetFunctions.hasOwnProperty(fn.toLowerCase()) ? applyFunction(fn, args) : match
+  );
+};
+
+const range = (start, end) =>
+  start > end ? [] : [start].concat(range(start + 1, end));
+
+const charRange = (start, end) => range(start.charCodeAt(0), end.charCodeAt(0));
+
+/*
+Use the `map` method and the `String.fromCharCode` function to convert each charCode in `charRange` into a character.
```

Comment:
This could be broken up into two steps, the first for chaining `map` to `range(start.charCodeAt(0), end.charCodeAt(0))` with `x` as a parameter and returning `x`. The next step could be passing `x` to `String.fromCharCode`.